src/app/Core/api.service.ts

```diff
@@ -0,0 +1,20 @@
+export interface ApiResponse<T>{
+    apiStatus: number;
+    apiProp?: T;
+}
+
+export async function ApiService(apiUrl: string): Promise<ApiResponse<Record<string, unknown>[]>> {
+    const apiResponse: Response = await fetch(apiUrl);
+
+    const apiStatus = apiResponse.status;
+
+    if (apiStatus === 200) {
+        const apiProp: Record<string, unknown>[] = await apiResponse.json();

```

Comment:
So here it would be: `const apiProp: T = await apiResponse.json()`

Reply:
Done.